packages/framework/presence/src/test/notificationsManager.spec.ts

```diff
@@ -3,16 +3,268 @@
  * Licensed under the MIT License.
  */
 
-import type { ISessionClient } from "../index.js";
-import { Notifications } from "../index.js";
+import { strict as assert } from "node:assert";
+
+import { EventAndErrorTrackingLogger } from "@fluidframework/test-utils/internal";
+import { useFakeTimers, type SinonFakeTimers, spy } from "sinon";
+
+import { Notifications, type ISessionClient } from "../index.js";
+import type { IEphemeralRuntime } from "../internalTypes.js";
 import type { IPresence } from "../presence.js";
+import type { createPresenceManager } from "../presenceManager.js";
+
+import { MockEphemeralRuntime } from "./mockEphemeralRuntime.js";
+import { assertFinalExpectations, prepareConnectedPresence } from "./testUtils.js";
+
+/**
+ * A map of presence signals used for testing, indexed by a friendly string.
+ */
+const knownSignals = new Map<string, Parameters<IEphemeralRuntime["submitSignal"]>>([
+	[
+		"PresenceJoin",
+		[
+			"Pres:DatastoreUpdate",
+			{
+				"sendTimestamp": 1010,
+				"avgLatency": 10,
+				"isComplete": true,
+				"data": {
+					"system:presence": {
+						"clientToSessionId": {
+							"client2": { "rev": 0, "timestamp": 1000, "value": "sessionId-2" },
+						},
+					},
+				},
+			},
+		],
+	],
+
+	[
+		"setNewId42",
+		[
+			"Pres:DatastoreUpdate",
+			{
+				"sendTimestamp": 1020,
+				"avgLatency": 10,
+				"data": {
+					"system:presence": {
+						"clientToSessionId": {
+							"client2": { "rev": 0, "timestamp": 1000, "value": "sessionId-2" },
+						},
+					},
+					"n:name:testNotificationWorkspace": {
+						"testEvents": {
+							"sessionId-2": {
+								"rev": 0,
+								"timestamp": 0,
+								"value": { "name": "newId", "args": [42] },
+								"ignoreUnmonitored": true,
+							},
+						},
+					},
+				},
+			},
+		],
+	],
+]);
 
 describe("Presence", () => {
 	describe("NotificationsManager", () => {
-		/**
-		 * See {@link checkCompiles} below
-		 */
-		it("API use compiles", () => {});
+		// Note: this test setup mimics the setup in src/test/presenceManager.spec.ts
+		let runtime: MockEphemeralRuntime;
+		let logger: EventAndErrorTrackingLogger;
+		const initialTime = 1000;
+		let clock: SinonFakeTimers;
+		let presence: ReturnType<typeof createPresenceManager>;
+
+		beforeEach(() => {});
+
+		before(async () => {
+			clock = useFakeTimers();
+		});
+
+		beforeEach(() => {
+			logger = new EventAndErrorTrackingLogger();
+			runtime = new MockEphemeralRuntime(logger);
+
+			// TODO: runtime.connected has been hacked in past to lie about true connection.
+			// This will need to be updated to an alternate status provider.
+			// runtime.connected = true;
```

Comment:
This hasn't been necessary in my testing.

Reply:
Update: You don't need this because `prepareConnectedPresence` does it.